### PR TITLE
Ericbrehault/sc 9807/popup mode for the widget

### PR DESCRIPTION
**Pull request by ebrehault**

### CODE REVIEW

File: libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  import type { KBStates } from '@nuclia/core';
+
+  export let backend = 'https://nuclia.cloud/api';
+  export let zone = 'europe-1';
+  export let knowledgebox: string;
+  export let placeholder = '';
+  export let lang = '';
+  export let cdn = '';
+  export let apikey = '';
+  export let kbslug = ''; // TODO: kbslug not needed anymore once regional system come into service
```
Comment (mpellerin42) [inline libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte:19]: regional system is there now, should we remove this already?

File: libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  onMount(() => {
+    const firstTry = enablePopupTrigger();
+    if (!firstTry) {
+      setTimeout(() => {
+        const secondTry = enablePopupTrigger();
+        if (!secondTry) {
+          console.info(
+            `No button found to open Nuclia’s search widget. Make sure you added 'data-nuclia="search-widget-button"' to a clickable element.`,
```
Comment (mpellerin42) [inline libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte:63]: No button found to open Nuclia’s search widget. Make sure you added 'data-nuclia="search-widget-button"' **attribute** to a clickable element.

File: libs/common/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -1235,6 +1235,8 @@
 	"widget.generator.advanced.display-category.hide-thumbnails.help": "Masquer les vignettes dans les résultats de recherche",
 	"widget.generator.advanced.display-category.hide-thumbnails.label": "Masquer les vignettes",
 	"widget.generator.advanced.display-category.placeholder": "Espace réservé dans la barre de recherche",
+	"widget.generator.advanced.display-category.popup-mode.help": "Affiche ls widget de recherche dans une fenêtre popup.",
```
Comment (mpellerin42) [inline libs/common/src/assets/i18n/fr.json:1238]: typo: "ls" => "la"

File: libs/common/src/lib/search-widget/widgets/widget-generator.component.ts [hunk 4]
```diff
@@ -669,13 +674,22 @@ export class WidgetGeneratorComponent implements OnInit, OnDestroy {
...
   knowledgebox="${kb.id}"
   ${zone}
-  features="${this.features}" ${ragProperties}${ragImagesProperties}${placeholder}${notEnoughDataMessage}${askToResource}${generativeModel}${filters}${preselectedFilters}${privateDetails}${backend}></nuclia-search-bar>
+  features="${this.features}" ${ragProperties}${ragImagesProperties}${placeholder}${notEnoughDataMessage}${askToResource}${generativeModel}${filters}${preselectedFilters}${privateDetails}${backend}></${tagName}>`;
+      if (!this.popupModeEnabled) {
+        baseSnippet += `
 <nuclia-search-results ${mode}></nuclia-search-results>`;
+      } else {
+        baseSnippet += `
+<div data-nuclia="search-widget-button">Click here to open the Nuclia search widget</div>`;
```
Comment (mpellerin42) [inline libs/common/src/lib/search-widget/widgets/widget-generator.component.ts:687]: why did you make a div instead of a button or a link (something semantically clickable)?

**Review by mpellerin42 — CHANGES_REQUESTED** [review]

File: libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte (hunk 2 above)
Comment (ebrehault) [reply libs/search-widget/src/widgets/popup-widget/PopupWidget.svelte:63]: debatable, the attribute is `data-nuclia` (and `"search-widget-button"` is its value), while `'data-nuclia="search-widget-button"'` is not an attribute, it's a just a snippet of code.

File: libs/common/src/lib/search-widget/widgets/widget-generator.component.scss [hunk 5]
```diff
@@ -151,7 +151,7 @@
         height: $content-height;
         overflow: auto;
         padding: rhythm(2) $padding-right-page rhythm(4) rhythm(1);
-        ::ng-deep div[data-nuclia='search-widget-button'] {
+        ::ng-deep a[data-nuclia='search-widget-button'] {
```
Comment (mpellerin42) [inline libs/common/src/lib/search-widget/widgets/widget-generator.component.scss:154]: if you didn't change your div, why did you change this style?

**Review by mpellerin42 — APPROVED** [review]
